source/particle/generator/probability_density_function.cc

```diff
@@ -0,0 +1,295 @@
+            particles.insert(std::make_pair(select_cell, new_particle));
+          }
+
+        return particles;
+      }
+
+
+
+      template <int dim>
+      void
+      ProbabilityDensityFunction<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.declare_entry ("Number of tracers", "1000",
+                               Patterns::Double (0),
+                               "Total number of tracers to create (not per processor or per element). "
+                               "The number is parsed as a floating point number (so that one can "
+                               "specify, for example, '1e4' particles) but it is interpreted as "
+                               "an integer, of course.");
+
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Random function");
+              {
+                Functions::ParsedFunction<dim>::declare_parameters (prm, 1);
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+
+
+      template <int dim>
+      void
+      ProbabilityDensityFunction<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            n_tracers    = static_cast<types::particle_index>(prm.get_double ("Number of tracers"));
+
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Random function");
+              {
+                try
+                  {
+                    function.parse_parameters (prm);
+                  }
+                catch (...)
+                  {
+                    std::cerr << "ERROR: FunctionParser failed to parse\n"
+                              << "\t'Initial conditions.Function'\n"
+                              << "with expression\n"
+                              << "\t'" << prm.get("Function expression") << "'";
+                    throw;
+                  }
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      ASPECT_REGISTER_PARTICLE_GENERATOR(ProbabilityDensityFunction,
+                                         "probability density function",
+                                         "Generate random distribution of "
+                                         "particles over entire simulation domain. "
+                                         "The particle density is prescribed in the "
```

Comment:
it's actually not a particle density you are describing, but just a (non-normalized) probability density. The two are proportional to each other, but not the same.